authentik/core/api/users.py

```diff
@@ -83,7 +104,7 @@ class UserSerializer(ModelSerializer):
     groups = PrimaryKeyRelatedField(
         allow_empty=True, many=True, source="ak_groups", queryset=Group.objects.all()
     )
-    groups_obj = ListSerializer(child=GroupSerializer(), read_only=True, source="ak_groups")
+    groups_obj = ListSerializer(child=SimpleGroupSerializer(), read_only=True, source="ak_groups")
```

Comment:
Please use `GroupMemberSerializer` in `groups.py` instead of creating a new serailizer

Reply:
How should the `GroupMemberSerializer` be utilized here? The `groups_obj` resolves the assigned groups (Using the `GroupSerializer`), which in turn resolve all it's users via the `GroupMemberSerializer`.

But option of replacing the `UserSerializer` is also not valid, or do I miss something here?

Reply:
the name `GroupMemberSerializer` is a bit misleading, `GroupMemberSerializer` is a stripped down version of the GroupSerializer without a lot of the extra fields and is very close to the serializer you've added, so re-using the same serializer is a bit cleaner imo

Reply:
We're talking about this one, right?

https://github.com/goauthentik/authentik/blob/main/authentik/core/api/groups.py#L17-L37

I tend do disagree, as it clearly extracts user information, as written in the docs as well!
`"""Stripped down user serializer to show relevant users for groups"""`

It is a stripped down version of the `UserSerializer`, not the `GroupSerializer`!

I took inspiration when creating the Serializer from it. In other terms I fully agree not to introduce new Serializers unless necessary. I have also written options within the issue this PR is referring to. The smaller GroupSerializer was actually the one with minimal impact on the existing codebase.

The LDAP outpost uses the groups object of the users to get the actual names when creating the entries. Stripping the `groups_obj` from the `UserSerializer` would require a code change there as well including additional requests from the LDAP outpost the retrieve the group names.

Reply:
Oh yeah my bad, I completely mis-remembered, disregard what I said above